### ISSUE TITLE
Investigate support for nio-transport-services

### ISSUE THREAD

**Issue by artemredkin** [swift-server/async-http-client#127]

**Comment by weissi** [comment]
cross posting from #135 

how this works is: You only need one `#if ...` and then you can

```swift
#if canImport(Network)
// Creating the "universal bootstrap" with the `NIOSSLClientTLSProvider`.
let tlsProvider = NIOTSClientTLSProvider(tlsOptions: .init())
let bootstrap = NIOTSClientTLSProvider(NIOTSEventLoopGroup(group: group), tls: tlsProvider)
#else
// TLS setup.
let configuration = TLSConfiguration.forClient()
let sslContext = try NIOSSLContext(configuration: configuration)

// Creating the "universal bootstrap" with the `NIOSSLClientTLSProvider`.
let tlsProvider = NIOSSLClientTLSProvider<ClientBootstrap>(context: sslContext, serverHostname: "example.com")
let bootstrap = NIOClientTCPBootstrap(ClientBootstrap(group: group), tls: tlsProvider)
#endif

// Bootstrapping a connection using the "universal bootstrapping mechanism"
let connection = bootstrap.enableTLS()
                     .connect(host: "example.com", port: 12345)
                     .wait()
```

For AsyncHTTPClient, you'll want to cache two bootstraps:

```
let tlsBootstrap: NIOClientTCPBootstrap = makeBootstrapForPlatform().enableTLS()
let plainTextBootstrap: NIOClientTCPBootstrap = makeBootstrapForPlatform()
```

and then you can just use the right one to bootstrap connections from depending on if that's TLS or not.

**Comment by artemredkin** [comment]
This was implemented in #184
